### PR TITLE
feat: no issue id in url

### LINKED ISSUE #1424: Feature: Remove url dependency for issue number
### Type of feature

🍕 Feature

### Current behavior

When visiting a repository it is required to have the goal issue number in the url. I would like to remove that so anyone can view any repo. 

### Suggested solution

Remove :id here https://github.com/open-sauced/open-sauced/blob/main/src/containers/Dashboard.jsx#L11

Also watch for tests to not fail. 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by mtfoley**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

In response to issue #1424, this PR makes the following changes:

- pulls issue ID references and dependent components out of Repository.jsx
- updates state for whether to show any kind of fork button ("Fork" or "View Fork")
- updates Route in App.jsx not to look at issue number, and to allow viewing repos
 without being logged in, but still passing user through
- updates Route in Dashboard.jsx not to look at issue number
- updates ListGoals component not use goal number
- updates tests for ListGoals component, including having links without goal number

## Related Tickets & Documents

Resolves #1424 

## Mobile & Desktop Screenshots/Recordings

Repository, URL does not contain issue number, user logged out:
Screen Shot 2022-05-06 at 10 43 20 PM

Repository, URL does not contain issue number, user logged in:
Screen Shot 2022-05-06 at 10 43 40 PM

ListGoals, link does not contain issue number.
Screen Shot 2022-05-06 at 10 44 00 PM

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?
been a while

### CODE REVIEW

**Comment by 0-vortex** [comment]
LGTM, just some testing linter issues - have yet to test it :D

**Comment by mtfoley** [comment]
@nworiekingslee, here is the PR we talked about.

**Review by 0-vortex — APPROVED** [review]

**Comment by nworiekingslee** [comment]
> @nworiekingslee, here is the PR we talked about.

This looks good

**Comment by bdougie** [comment]
Thanks @mtfoley

File: src/containers/Dashboard.jsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ function Repositories({user, match}) {
   return (
     <Suspense fallback={<Spinner />}>
       <Route exact path={match.url} render={(props) => <RepositoryGoals user={user} {...props} />} />
-      <Route path={"/repos/:repoOwner/:repoName/:id"} render={(props) => <Repository user={user} {...props} />} />
+      <Route path={"/repos/:repoOwner/:repoName"} render={(props) => <Repository user={user} {...props} />} />
```
Comment (bdougie) [inline src/containers/Dashboard.jsx:11]: Curious if we should eventually remove "/repos." Not sure if that will have adverse affects, but worth considering.

**Review by bdougie — APPROVED** [review]

**Comment by bdougie** [comment]
Merging this now, and will start adding some insights and data soon.

**Pull request merged**